examples/tests/test_00_consumer.py

```diff
@@ -0,0 +1,129 @@
+"""
+Test the consumer with Pact.
+
+This module tests the consumer defined in `src/consumer.py` against a mock
+provider. The mock provider is set up by Pact, and is used to ensure that the
+consumer is making the expected requests to the provider, and that the provider
+is responding with the expected responses. Once these interactions are
+validated, the contracts can be published to a Pact Broker. The contracts can
+then be used to validate the provider's interactions.
+"""
+
+from __future__ import annotations
+
+import logging
+from http import HTTPStatus
+from typing import TYPE_CHECKING, Any, Generator
+
+import pytest
+import requests
+from examples.src.consumer import User, UserConsumer
+from pact import Consumer, Format, Like, Provider
+from yarl import URL
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+    from pact.pact import Pact
+
+log = logging.getLogger(__name__)
+
+MOCK_URL = URL("http://localhost:8080")
+
+
+@pytest.fixture()
+def user_consumer() -> UserConsumer:
+    """
+    Returns an instance of the UserConsumer class.
+
+    As we do not want to stand up all of the consumer's dependencies, we direct
+    the consumer to use Pact's mock provider. This allows us to define what
+    requests the consumer will make to the provider, and what responses the
+    provider will return.
+    """
+    return UserConsumer(str(MOCK_URL))
+
+
+@pytest.fixture(scope="module")
+def pact(broker: URL, pact_dir: Path) -> Generator[Pact, Any, None]:
+    """
+    Set up Pact.
+
+    In order to test the consumer in isolation, Pact sets up a mock version of
+    the provider. This mock provider will expect to receive defined requests
+    and will respond with defined responses.
+
+    The fixture here simply defines the Consumer and Provide, and sets up the
+    mock provider. With each test, we define the expected request and response
+    from the provider as follows:
+
+    ```python
+    pact.given("UserA exists and is not an admin") \
+        .upon_receiving("A request for UserA") \
+        .with_request("get", "/users/123") \
+        .will_respond_with(200, body=Like(expected))
```

Comment:
is it worth being explicit about the data structure returned by our API, prior to considering writing the test, as a JSON string, rather than a dict.

Considering our provider returns the following json response

`'{"id":1,"name":"foo", "created_on": "2020-07-10 15:00:00.000"}'`

We want to consider the minimal response required for our test/consumer to pass the scenario (so we don't include extraneous information) that we don't care if the provider returns or not.

Ie if our consumer didn't need created_on for one of its scenarios (it doesn't in this case, it always requires the three fields)

Reply:
I have addressed this more explicitly when defining the `expected` response later in the test. But good call to be pointing this out explicitly.